src/components/tools/Stats/MonthTiming.svelte

```diff
@@ -0,0 +1,131 @@
+<script>
+  import { SkeletonText } from "carbon-components-svelte";
+  import { timeFormat } from "d3-time-format";
+  import { rollups, merge } from "d3-array";
+  import StatBlock from "./StatBlock.svelte";
+
+  /** Array of data values grouped by date, e.g.
+  {
+    date:  Date Sat Dec 31 1949 16:00:00 GMT-0800 (Pacific Standard Time),
+    values: [
+      { id: "livneh", label: "Observed", value: 74.6},
+      { id: "CanESM2", label: "CanESM2 (Average)", value: 75.2},
+      { id: "CNRM-CM5", label: "CNRM-CM5 (Cool/Wet)", value: 73.3}
+    ]
+  }
+  */
+  export let data;
+
+  /** List of 1 or more groups
+   * e.g. modeled historical, modeled projections, observed historical */
+  export let groupList;
+
+  /** List of 1 or more periods
+  e.g. baseline, mid-century, end-century */
+  export let periodList;
+
+  /** Id of default group to initialize StatPanel. This prop is optional. Defaults to first item in groupList */
+  export let groupId;
+
+  /** Id of default period to initialize StatPanel. This prop is optional. Defaults to first item in periodList */
+  export let periodId;
+
+  /** List of selected models */
+  export let models;
+
+  export let units;
+
+  let metrics = [];
+
+  const dateFormat = timeFormat("%B");
+
+  function subsetByYears(start, end) {
+    return function (d) {
+      return (
+        d.date >= new Date(Date.UTC(start, 0, 1)) &&
+        d.date <= new Date(Date.UTC(end, 11, 31))
+      );
+    };
+  }
+
+  function getMonths(data) {
+    if (Array.isArray(data) && data.length) {
+      const months = new Set(data.map(({ date }) => date.getMonth()));
+      const sortedMonths = [...months].sort((a, b) => a - b);
+      return {
+        earliest: dateFormat(new Date(2020, sortedMonths[0], 1)),
+        latest: dateFormat(
+          new Date(2020, sortedMonths[sortedMonths.length - 1], 1)
```

Comment:
And here?

Reply:
I see that you're just formatting the date to get the month. Because you'll have a value representing 0–11 for January to December you could use that as an index with an array of month strings, like:

```js
const MONTH_NAMES_LONG = ["January", "February", "March", ...]
MONTH_NAMES_LONG[0];
```

Might be good to have that as a common `const` for the tools?

Otherwise I think you can just do something like `earliest: dateFormat((new Date()).setMonth(monthIndex))` if you're only ever using `earliest` as a string.

Reply:
Fixed via https://github.com/berkeley-gif/caladapt-website-2021/pull/108/commits/faba838a1d59f1f8b649f6f468a30fc3a678d534